procfile/procfile.go

```diff
@@ -57,3 +65,58 @@ func NewProcfileFromPath(path string) (Procfile, error) {
 
 	return p, nil
 }
+
+// NewProcfileFromBinding creates a Procfile by reading Procfile from bindings if it exists.  If it does not exist, returns an
+// empty Procfile.
+func NewProcfileFromBinding(binds libcnb.Bindings) (Procfile, error) {
+
+	p := Procfile{}
+	if binding, ok, err := bindings.ResolveOne(binds, bindings.OfType(BindingType)); err != nil {
+		return nil, fmt.Errorf("unable to resolve binding\n%w", err)
+	} else if ok {
+		if path, ok := binding.SecretFilePath(BindingType); ok {
+			if p, err = NewProcfileFromPath(filepath.Dir(path)); err != nil {
+				return nil, err
+			}
+			return p, nil
+		} else {
+			return nil, fmt.Errorf("unable to find Procfile from binding")
+		}
+	} else {
+		return p, nil
+	}
+}
+
+// NewProcfileFromPathOrBinding attempts to create a merged Procfile from given path and bindings.  If neither can be created, returns an
+// empty Procfile.
+func NewProcfileFromPathOrBinding(path string, binds libcnb.Bindings) (Procfile, error) {
+
+	procPath, procBind := Procfile{}, Procfile{}
```

Comment:
Is there a reason to define these separately? I think typically you just see variables defined as the result of the function:

```
procPath, err := NewProcfileFromPath(path);
if err != nill {
...
}
```

and the same for `procBind`. You don't seem to need the separate error variables either.

I don't see anything technically wrong with the code, but you might be able to shrink the function a bit and make it a little more idiomatic.

Reply:
✅ 
I think I originally had it a in way that needed this and just didn't change it back when I improved the logic 🤔